mantidproject/mantid
# Custom ParaView representation for axis-aligned vtkStructuredGrids

Description of work.

This adds a [custom representation](https://www.paraview.org/Wiki/ParaView/Plugin_HowTo#Adding_new_Representations_for_3D_View_using_Plugins_.2A_new_in_version_3.7) based on the `vtkGeometrySliceRepresentation`.  The main change is an `AlignedCutter` class which that speeds up slicing by taking advantage of the consistent bin sizes in a `MDHistoWorkspace`. Other data types like `vtkUnstructuredGrid` fall back to the implementation in `vtkCutter`.

Combined with the patch from [this MR](https://gitlab.kitware.com/paraview/paraview/merge_requests/1850), one should be close to fluid movement with smaller (200x200x200) datasets.
 
**To test:**



* Build the `master` branch with `MAKE_VATES=ON`, load a `MDHistoWorkspace` in the Multislice view. Drag the slices to different positions and observe how long it takes to draw each slice.
* Build this branch and draw load the same `MDHistoWorkspace`. Verify that the view is significantly more responsive.
* Load a `MDEventWorkspace` in the Multislice view and verify there are no issues with the data type.


There is no GitHub issue associated with this pull request.

**Release Notes** 
[here](https://github.com/mantidproject/mantid/pull/20303/commits/d678835b44c80afe87db94a5b297816f73211f1e)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
